src/main.ts

```diff
@@ -104,15 +95,32 @@ const crawler = new CheerioCrawler({
             pageReviews.push(pageReview);
         });
 
+        const reviewsNeeded = maxShopReviewsPerCrawl - state[shopName];
+        pageReviews = pageReviews.slice(0, Math.min(pageReviews.length, reviewsNeeded));
```

Comment:
`Math.min` is not needed here, slice will not add empty values

```suggestion
        pageReviews = pageReviews.slice(0, reviewsNeeded);
```

Reply:
fixed